cmd/setup.go

```diff
@@ -304,3 +308,39 @@ func configureLoadPoints(conf config, cp *ConfigProvider) (loadPoints []*core.Lo
 
 	return loadPoints, nil
 }
+
+func configureCircuits(site *core.Site, loadPoints []*core.LoadPoint, cp *ConfigProvider) (err error) {
+	ccInterfaces, _ := viper.AllSettings()["circuits"].([]interface{})
+
+	for _, ccI := range ccInterfaces {
+		var ccMap map[string]interface{}
+		if err := util.DecodeOther(ccI, &ccMap); err != nil {
+			return fmt.Errorf("failed decoding circuit configuration: %w", err)
+		}
+
+		ccNew, err := core.NewCircuitFromConfig(cp, ccMap, site)
+		if err != nil {
+			return fmt.Errorf("failed configuring circuit: %w", err)
+		}
+		site.Circuits = append(site.Circuits, ccNew)
+	}
+	// connect circuits and lps
+	for lpId := range loadPoints {
+		for ccId := range site.Circuits {
+			loadPoints[lpId].CircuitPtr = site.Circuits[ccId].GetCircuit(loadPoints[lpId].CircuitRef)
+			if loadPoints[lpId].CircuitPtr != nil {
+				// check there is a virtual meter, and add lp in case as consumer
+				if vmtr := loadPoints[lpId].CircuitPtr.GetVMeter(); vmtr != nil {
+					vmtr.AddConsumer(loadPoints[lpId])
+				}
+				break
+			}
+		}
+		if loadPoints[lpId].CircuitRef != "" && loadPoints[lpId].CircuitPtr == nil {
```

Comment:
gehört eigentlich oben in explizites Error return

Reply:
nein, da erst nach Iteration über alle LP den Fehler feststelle (Circtuits müssen keinen root circuit haben, es kann mehrere top level circuits geben)

Reply:
Kommentar hinzugefügt